docs/BOOT_LOADER_SPECIFICATION.md

```diff
@@ -187,102 +132,129 @@ the root of the file system. Specifically, if `$BOOT` is the ESP, then
-  example to show only a single newest kernel per OS, or to group items by OS,
-  or to maybe filter out the currently booted OS in UIs that want to show only
-  other installed operating systems. This ID shall be formatted as 32 lower
-  case hexadecimal characters (i.e. without any UUID formatting). This key is
-  optional. Example: `4098b3f648d74c13b1f04ccfba7798e8`.
-* `sort-key` shall contain a short string used for sorting entries on
-  display. This can be defined freely though should typically be initialized
-  from `IMAGE_ID=` or `ID=` from `/etc/os-release` of the relevant entry,
-  possibly suffixed. This field is optional. If set, it is used as primary
-  sorting key for the entries on display (lexicographically increasing). It
-  does not have to be unique (and usually is not). If non-unique the the
-  `machine-id` (lexicographically increasing) and `version` (lexicographically
-  decreasing, i.e. newest version first) fields described above are used as
-  secondary/ternary sorting keys. If this field is not set entries are
-  typically sorted by the `.conf` file name of the entry.
-* `linux` refers to the Linux kernel to spawn and shall be a path relative to
-  `$BOOT`. It is recommended that every distribution creates a machine id and
-  version specific subdirectory below `$BOOT` and places its kernels and
-  initial RAM disk images there. Example:
-  `/6a9857a393724b7a981ebb5b8495b9ea/3.8.0-2.fc19.x86_64/linux`.
-* `initrd` refers to the initrd to use when executing the kernel. This also
+set: ASCII upper and lower case characters, digits, "+", "-", "_" and ".".
+Also, the file names should have a length of at least one and at most 255
+characters (including the file name suffix).
+
+These configuration snippets shall be UNIX-style text files (i.e. lines
+separated by a single newline character), in the UTF-8 encoding. The
+configuration snippets are loosely inspired by Grub1's configuration syntax.
+Lines beginning with "#" are used for comments and shall be ignored. The first
+word of a line is used as key and is separated by one or more spaces from the
+value.
+
+#### Type #1 Boot Loader Entry Keys
+
+The following keys are recognized:
+
+* `title` is a human-readable title for this menu item to be displayed in the
+  boot menu. It is a good idea to initialize this from the `PRETTY_NAME=` of
+  `/etc/os-release`. This name should be descriptive and does not have to be
+  unique. If a boot loader discovers two entries with the same title it should
+  show more than just the raw title in the UI, for example by appending the
+  `version` field. This field is optional.
+
+  Example: `title Fedora 18 (Spherical Cow)`
+
+* `version` is a human-readable version for this menu item. This is usually the
+  kernel version and is intended for use by OSes to install multiple kernel
+  versions with the same `title` field. This field shall be in a syntax that is
+  useful for Debian-style version sorts, so that the boot loader UI can
```

Comment:
Funny you mention debian-style version sort. Afaik our version compare isn't fully debian compatible (#4641).

Reply:
That was in the original text, which I only move around here. This text is later removed. (I think it'd be maybe worth it to add a subsection that enumerates differences to rpm and deb version comparisons, since we're not following either exactly.)